ni_measurementlink_service/session_management.py

```diff
@@ -418,6 +444,520 @@ def create_sessions(
         """
         return self._create_sessions_core(session_constructor, instrument_type_id)
 
+    @requires_feature(SESSION_MANAGEMENT_2024Q1)
+    def create_nidcpower_session(
+        self,
+        reset: bool = False,
+        options: Optional[Dict[str, Any]] = None,
+        initialization_behavior: SessionInitializationBehavior = SessionInitializationBehavior.AUTO,
+    ) -> ContextManager[TypedSessionInformation[nidcpower.Session]]:
+        """Create a single NI-DCPower instrument session.
```

Comment:
Since this is the main API for creating sessions, I would make sure the documentation mentions that if multiple sessions are created, an exception is thrown.

Reply:
Same comment applies to other drivers.

Reply:
I added a `Raises:` section:
```
PS D:\dev\measurementlink-python> rg "ValueError: If" -C 1
ni_measurementlink_service\session_management.py
419-        Raises:
420:            ValueError: If the instrument type ID is empty, no reserved sessions
421-                match the instrument type ID, or too many reserved sessions
--
450-        Raises:
451:            ValueError: If the instrument type ID is empty or no reserved
452-                sessions match the instrument type ID.
--
482-        Raises:
483:            ValueError: If no NI-DCPower sessions are reserved or too many
484-                NI-DCPower sessions are reserved.
--
523-        Raises:
524:            ValueError: If no NI-DCPower sessions are reserved.
525-
--
562-        Raises:
563:            ValueError: If no NI-Digital sessions are reserved or too many
564-                NI-Digital sessions are reserved.
--
607-        Raises:
608:            ValueError: If no NI-Digital sessions are reserved.
609-
--
650-        Raises:
651:            ValueError: If no NI-DMM sessions are reserved or too many
652-                NI-DMM sessions are reserved.
--
695-        Raises:
696:            ValueError: If no NI-DMM sessions are reserved.
697-
--
738-        Raises:
739:            ValueError: If no NI-FGEN sessions are reserved or too many NI-FGEN
740-                sessions are reserved.
--
783-        Raises:
784:            ValueError: If no NI-FGEN sessions are reserved.
785-
--
826-        Raises:
827:            ValueError: If no NI-SCOPE sessions are reserved or too many
828-                NI-SCOPE sessions are reserved.
--
871-        Raises:
872:            ValueError: If no NI-SCOPE sessions are reserved.
873-
--
919-        Raises:
920:            ValueError: If no relay driver sessions are reserved or
921-                too many relay driver sessions are reserved.
--
970-        Raises:
971:            ValueError: If no relay driver sessions are reserved.
972-
```

I also updated the errors to say "No *reserved* sessions" or "Too many *reserved* sessions" and updated the NI-SWITCH method description to say "NI-SWITCH relay driver sessions" (to distinguish from multiplexer).